### PR TITLE
[114] OnboardArea instanciated even when a representation is open

### PR DESCRIPTION
**Pull request by SteveMonnier**
If the browser is reloaded then RepresentationArea.tsx refresh will not
cause to reload the OnboardArea.

Bug: https://github.com/eclipse-sirius/sirius-components/issues/114
Signed-off-by: Steve Monnier <steve.monnier@obeo.fr>

### Type of this PR 

- [ ] Bug fix
- [ ] New feature or improvement
- [ ] Documentation
- [ ] Cleanup
- [ ] Test
- [ ] Build/Releng

### Issue(s)

...

### What does this PR do?

...

### Screenshot/screencast of this PR

...
 

### Potential side effects

...

### How to test this PR?

- [ ] Frontend Unit tests
- [ ] Backend Unit tests
- [ ] Cypress : please specify
- [ ] Manual Test : please specify

### Checklist

- [ ] I have read CONTRIBUTING carefully.
- [ ] I have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [ ] All my commits are signed-off (`-s`) with my mail address of my Eclipse Account.
- [ ] I have covered my changes by unit tests or integration tests or manual tests.
- [ ] All tests pass.
- [ ] I have updated the documentation accordingly
- [ ] New React components are availables in storybook

### CODE REVIEW

File: frontend/src/views/edit-project/RepresentationArea.tsx [hunk 1]
```diff
@@ -36,7 +36,10 @@ export const RepresentationArea = ({
 }) => {
   let content;
   if (!displayedRepresentation) {
-    content = <OnboardArea selection={selection} setSelection={setSelection} />;
+    if (window.performance && performance.navigation.type !== PerformanceNavigation.TYPE_RELOAD) {
```
Comment (sbegaudeau) [inline frontend/src/views/edit-project/RepresentationArea.tsx:39]: As discussed during the meeting, the issue is a matter of wrong lifecycle in our state machines. Do not try to detect reload events. On top of that you are relying on a [deprecated API](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigation/type)

**Review by sbegaudeau — DISMISSED** [review]

**Comment by pcdavid** [comment]
From what I understand the issue is in `EditProjectView`. When it is rendered for a specific representation, we get the representation id from `const { projectId, representationId } = useParams();`, and launch
```js
getRepresentation({ projectId: project.id, representationId });
```
in a `useEffect()`. At this point in time:
* we know we will display a particular representation
* we have asked the server to send it to us
* we have not yet received the response

There should be a new intermediate state, something like `PROJECT_LOADED_AND_REPRESENTATION_LOADING__STATE`. While we are in this state, the Onboarding component should not be instanciated, we know it will be replaced by the actual representation as soon as we get it.

I've added some comments below at the locations where I believe the new state and its transition should be done.

```js
  const [getRepresentation, result] = useLazyQuery(getRepresentationQuery, {}, 'getRepresentation');
  useEffect(() => {
    if (representationId && project && !displayedRepresentation) {
      getRepresentation({ projectId: project.id, representationId });
      // Transition to a new state: we know we *will* display an actual representation, but we have not received it from the backend
      // In this new state, Onboarding should *not* be instanciated, we know it will be replaced by the actual representation as soon as we get it.
    }
  }, [project, representationId, displayedRepresentation, getRepresentation]);
  useEffect(() => {
    if (!result.loading && result?.data?.data?.viewer?.project?.representation) {
      // The `getRepresentation` query from above has finished, we now have the actual representation we wanted to display
      // We can transition to the normal "PROJECT_LOADED_AND_REPRESENTATION_DISPLAYED__STATE"
      const { id, label, __typename } = result.data.data.viewer.project.representation;
      const representation = { id, label, kind: __typename };
      const action = { type: HANDLE_SELECTION__ACTION, selection: representation };
      dispatch(action);
    }
  }, [result]);
```

**Comment by pcdavid** [comment]
> ```js
>   const [getRepresentation, result] = useLazyQuery(getRepresentationQuery, {}, 'getRepresentation');
>   useEffect(() => {
>     if (representationId && project && !displayedRepresentation) {
>       getRepresentation({ projectId: project.id, representationId });
>       // Transition to a new state: we know we *will* display an actual representation, but we have not received it from the backend
>       // In this new state, Onboarding should *not* be instanciated, we know it will be replaced by the actual representation as soon as we get it.
>     }
>   }, [project, representationId, displayedRepresentation, getRepresentation]);
> ```

Actually it's probably too late here to "switch" to the new state. The "LOADING" state should be the first state of the component *if* we know frmo the URL that we are on a specific representation.

File: frontend/src/views/edit-project/RepresentationArea.tsx [hunk 2]
```diff
@@ -36,7 +36,10 @@ export const RepresentationArea = ({
 }) => {
   let content;
   if (!displayedRepresentation) {
-    content = <OnboardArea selection={selection} setSelection={setSelection} />;
+    if (window.performance) {
+      // If the user reloads a diagram page, it should load the OnboardArea
+      content = <OnboardArea selection={selection} setSelection={setSelection} />;
+    }
```
Comment (pcdavid) [inline frontend/src/views/edit-project/RepresentationArea.tsx:42]: Leftover from previous version? This API should not be used (and it should not be necessary to fix the issue).

**Review by pcdavid — CHANGES_REQUESTED** [review]

File: frontend/src/views/edit-project/RepresentationArea.tsx (hunk 2 above)
Comment (SteveMonnier) [reply frontend/src/views/edit-project/RepresentationArea.tsx:42]: Indeed I forgot to remove this "if"

File: frontend/src/views/edit-project/RepresentationArea.tsx [hunk 3]
```diff
@@ -41,6 +41,7 @@ export const RepresentationArea = ({
 }) => {
   let content;
   if (!displayedRepresentation) {
+    // If the user reloads a diagram page, it should load the OnboardArea
```
Comment (pcdavid) [inline frontend/src/views/edit-project/RepresentationArea.tsx:44]: The comment seems obsolete/irrelevant/wrong. I'll remove it.

**Review by pcdavid — APPROVED** [review]
I'll make a final test and if it's OK rebase and remove the invalid comment and merge.

**Review by pcdavid — APPROVED** [review]

**Pull request merged**